lib/charms/mongodb_libs/v0/mongodb_provider.py

```diff
@@ -203,7 +206,7 @@ def _get_roles_from_relation(self, relation: Relation) -> Set[str]:
             if unit.app is self.charm.app:
                 # it is peer relation, skip
                 continue
-            roles = relation.data[unit].get("extra-user-roles", None)
+            roles = relation.data[relation.app].get("extra-user-roles", None)
```

Comment:
in such a case, we don't need to iterate via `relation.units` can you please remove lines 202-205

Reply:
Gotcha! Great catch, thanks @delgod